extensions/blocks/send-a-message/whatsapp-button/edit.js

```diff
@@ -0,0 +1,260 @@
+/**
+ * External dependencies
+ */
+import { __, _x } from '@wordpress/i18n';
+import { useEffect, useState } from '@wordpress/element';
+import { withDispatch } from '@wordpress/data';
+import { compose } from '@wordpress/compose';
+import {
+	Button,
+	BaseControl,
+	TextControl,
+	TextareaControl,
+	SelectControl,
+	PanelBody,
+	ToolbarGroup,
+	Dropdown,
+	Path,
+} from '@wordpress/components';
+import {
+	BlockControls,
+	InspectorControls,
+	RichText,
+	PanelColorSettings,
+} from '@wordpress/block-editor';
+import { DOWN } from '@wordpress/keycodes';
+
+/**
+ * Internal dependencies
+ */
+import { countryCodes } from '../shared/countrycodes.js';
+import renderMaterialIcon from '../../../shared/render-material-icon';
+import HelpMessage from '../../../shared/help-message';
+import './view.scss';
+
+const WHATSAPP_GREEN = '#25D366';
+const WHATSAPP_DARK = '#465B64';
+const WHATSAPP_LIGHT = '#F4F4F4';
+
+const WhatsAppButtonEdit = ( { attributes, setAttributes, className, selectBlock, clientId } ) => {
+	const {
+		countryCode,
+		phoneNumber,
+		buttonText,
+		firstMessage,
+		colorClass,
+		backgroundColor,
+	} = attributes;
+
+	const [ isValidPhoneNumber, setIsValidPhoneNumber ] = useState( true );
+
+	const getCountryCode = async () => {
+		setAttributes( { countryCode: '1' } );
+
+		const geoFetch = await fetch( 'http://ip-api.com/json/?fields=countryCode' )
+			.then( response => {
+				if ( ! response.ok ) {
+					return false;
+				}
+
+				return response;
+			} )
+			.catch( () => {
+				return false;
+			} );
+
+		if ( geoFetch ) {
+			const geo = await geoFetch.json();
+
+			countryCodes.forEach( item => {
+				if ( item.code === geo.countryCode ) {
+					setAttributes( { countryCode: item.value } );
+				}
+			} );
+		}
+	};
+
+	useEffect( () => {
+		if ( undefined === countryCode ) {
+			getCountryCode();
+			selectBlock( clientId );
+		}
+	} );
```

Comment:
⚠️ Overkill suggestion alert! ⚠️ 

Without dependencies this would run at every render, while we only want to check on `countryCode` change.
The complete list of deps is `[clientId, countryCode, getCountryCode, selectBlock]`, which also entails wrapping `getCountryCode` in a `useCallback` hook.

In reality, `getCountryCode` only seems to run once, so... 🤷